src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -48,8 +49,27 @@
 class AzCliHelp(CLIHelp):
 
     def __init__(self, cli_ctx):
-        self.cli_ctx = cli_ctx
-        super(AzCliHelp, self).__init__(cli_ctx, PRIVACY_STATEMENT, WELCOME_MESSAGE)
+        super(AzCliHelp, self).__init__(cli_ctx,
+                                        privacy_statement=PRIVACY_STATEMENT,
+                                        welcome_message=WELCOME_MESSAGE,
+                                        command_help_cls=CliCommandHelpFile,
+                                        help_cls=CliHelpFile)
+        from knack.help import HelpObject
+
+        # TODO: This workaround is used to avoid a bizarre bug in Python 2.7. It
+        # essentially reassigns Knack's HelpObject._normalize_text implementation
+        # with an identical implemenation in Az. For whatever reason, this fixes
+        # the bug in Python 2.7.
+        @staticmethod
+        def new_normalize_text(s):
+            if not s or len(s) < 2:
+                return s or ''
+            s = s.strip()
+            initial_upper = s[0].upper() + s[1:]
+            trailing_period = '' if s[-1] in '.!?' else '.'
+            return initial_upper + trailing_period
+
+        HelpObject._normalize_text = new_normalize_text  # pylint: disable=protected-access
```

Comment:
Can you explain what the bug was that caused this to be needed?

Reply:
I think this is just ensuring that all help will end in a proper "stopping" punctuation mark.

Reply:
It shouldn't be necessary and works just fine without this on Python 3. But for whatever reason, stupid Python 2.7 ends up looking into the SDK docstring, finds some non-ASCII character, and crashes. Not wanting to invest a lot of time into the philosophical question of why Python 2.7 is such crap, I just went with this.